www/js/app.js

```diff
@@ -2303,6 +2318,112 @@ function articleLoadedSW (iframeArticleContent) {
     params.isLandingPage = false;
 };
 
+/**
+ * Attaches popover trigger events to the given window
+ * @param {Window} win The window to which to attach popover trigger events
+ */
+function attachPopoverTriggerEvents (win) {
+    // The popover feature requires as a minimum that the browser supports the css matches function
+    // (having this condition prevents very erratic popover placement in IE11, for example, so the feature is disabled for such browsers)
+    if (appstate.wikimediaZimLoaded && params.showPopoverPreviews && 'matches' in Element.prototype) {
+        const iframeDoc = win.document;
+        if (iframeDoc) {
+            // Attach the popover CSS to the current article document
+            uiUtil.attachKiwixPopoverCss(iframeDoc);
+            // Add event listeners to iframe window to check for links
+            win.addEventListener('mouseover', handlePopoverEvents, true);
+            win.addEventListener('focus', handlePopoverEvents, true);
+            // Conditionally add event listeners to support touch events with fallback to pointer events
```

Comment:
Aren't these proxies for "click" events? Are we showing the popover when someone clicks or only when they hover/focus?

Reply:
Not really. There is some tricky event handling in this feature, due to the need to support: 1. Mouse-based devices; 2. Touchscreen laptops and mobile; 3. Keyboard. To be clear:

1. These event listeners are added to the article's window (not to each link individually) - the code looks for the hovered / touched element and only executes if it is an Anchor;
2. `mouseover` (hovering over a link without clicking it), `touchstart` and `pointerdown` will all show the popover if the user hovers / touches / holds mouse down for longer than 600ms (this is a timeout defined in `uiUtil.js`). We only listen to `pointerdown` if `touchstart` is not available in the browser (e.g. the UWP framework, which uses Edge Legacy, supports touch, but not Touch events, only Pointer events);
3. `focus` supports keyboard users, who may tab into a link, but have not "clicked" it (usually achieved by using spacebar after a link is focused). Note that hovering does not focus an element, and `touchstart` / `pointerdown` occur before focus in all browsers. In some only the touch events will fire and not focus. In others, both events will fire, but because touch/pointer occurs first, the second `focus` activation gets cancelled due to flags set.

Reply:
Ah okay I missed that they were being added to the window. Sounds good.